flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/endpoint/hive/util/OperationExecutorFactory.java

```diff
@@ -203,6 +230,78 @@ private static ResultSet executeGetTables(
                         .collect(Collectors.toList()));
     }
 
+    private static ResultSet executeGetFunctions(
+            SqlGatewayService service,
+            SessionHandle sessionHandle,
+            @Nullable String catalogName,
+            @Nullable String databasePattern,
+            @Nullable String functionNamePattern) {
+        String specifiedCatalogName =
+                isNullOrEmpty(catalogName) ? service.getCurrentCatalog(sessionHandle) : catalogName;
+
+        Set<FunctionInfo> candidates = new HashSet<>();
+        // Add user defined functions
+        for (String databaseName :
+                filter(
+                        service.listDatabases(sessionHandle, specifiedCatalogName),
+                        databasePattern)) {
+            candidates.addAll(
+                    service.listUserDefinedFunctions(
+                            sessionHandle, specifiedCatalogName, databaseName));
+        }
+        // Add system functions
+        if (isNullOrEmpty(catalogName) && isNullOrEmpty(databasePattern)) {
+            candidates.addAll(service.listSystemFunctions(sessionHandle));
+        }
+        // Filter out unmatched functions
+        Set<FunctionInfo> matchedFunctions =
+                filter(
+                        candidates,
+                        candidate -> candidate.getIdentifier().getFunctionName(),
+                        functionNamePattern);
+        return new ResultSet(
+                EOS,
+                null,
+                GET_FUNCTIONS_SCHEMA,
```

Comment:
Maybe you can add buildResultSet method similar to HiveServer2Schemas#buildSchema since the result set always contains 'EOS' and 'null'.